### PR TITLE
Optional bigint support

### PR DESCRIPTION
**Pull request by tristanmenzel**
Revisiting [PR401](https://github.com/bcherny/json-schema-to-typescript/pull/401) to add bigint support. 

Hopefully this PR addresses the outstanding comments on the original PR.

### CODE REVIEW

File: src/parser.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export function parse(
     return parseLiteral(schema, keyName)
   }
 
-  const types = typesOfSchema(schema)
+  const types = typesOfSchema(schema, options)
```
Comment (bcherny) [inline src/parser.ts:48]:
```suggestion
  const types = typesOfSchema(schema)
```

File: src/types/JSONSchema.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ export type SchemaType =
   | 'NEVER'
   | 'NULL'
   | 'NUMBER'
+  | 'BIGINT'
```
Comment (bcherny) [inline src/types/JSONSchema.ts:15]: Nit: alphabetize please.

File: src/utils.ts [hunk 3]
```diff
@@ -49,6 +49,16 @@ const BLACKLISTED_KEYS = new Set([
   'not'
 ])
 
+// Determines if the constraints on a given integer schema could result in values that are unsafe for the JS Number type
+export function isUnsafeIntegerForNumberType({minimum, maximum}: {minimum?: number; maximum?: number}) {
```
Comment (bcherny) [inline src/utils.ts:53]:
```suggestion
export function isUnsafeIntegerForNumberType({minimum, maximum}: JSONSchema4) {
```

File: test/e2e/bigint.1.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    unsafeSizedNumber: {
+      type: 'integer',
+      minimum: -9007199254740992,
+      maximum: 9007199254740992
+    }
+  }
+}
+
+export const options = {
+  ...DEFAULT_OPTIONS,
```
Comment (bcherny) [inline test/e2e/bigint.1.ts:30]:
```suggestion
```

File: test/e2e/bigint.1.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import {DEFAULT_OPTIONS} from '../../src'
+
```
Comment (bcherny) [inline test/e2e/bigint.1.ts:2]:
```suggestion
```

File: test/e2e/bigint.2.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    unsafeSizedNumber: {
+      type: 'integer',
+      minimum: -9007199254740992,
+      maximum: 9007199254740992
+    }
+  }
+}
+
+export const options = {
+  ...DEFAULT_OPTIONS,
```
Comment (bcherny) [inline test/e2e/bigint.2.ts:30]:
```suggestion
```

File: test/e2e/bigint.2.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import {DEFAULT_OPTIONS} from '../../src'
```
Comment (bcherny) [inline test/e2e/bigint.2.ts:1]:
```suggestion
```

File: src/cli.ts [hunk 8]
```diff
@@ -184,6 +184,8 @@ Boolean values can be set to false using the 'no-' prefix.
       Output unknown type instead of any type
   --unreachableDefinitions
       Generates code for definitions that aren't referenced by the schema
+  --enableBigInt
```
Comment (bcherny) [inline src/cli.ts:187]: Nit: Alphabetize please.

File: src/generator.ts [hunk 9]
```diff
@@ -188,6 +188,8 @@ function generateRawType(ast: AST, options: Options): string {
       return 'never'
     case 'NUMBER':
       return 'number'
+    case 'BIGINT':
```
Comment (bcherny) [inline src/generator.ts:191]: Nit: Alphabetize please.

File: src/index.ts [hunk 10]
```diff
@@ -76,6 +76,10 @@ export interface Options {
    * Generate unknown type instead of any
    */
   unknownAny: boolean
+  /**
+   * Use `bigint` for unbounded (i.e. no 'maximum') `integer` in the schema.
+   */
+  enableBigInt: boolean
```
Comment (bcherny) [inline src/index.ts:82]: Nit: Alphabetize please.

File: src/index.ts [hunk 11]
```diff
@@ -104,7 +108,8 @@ export const DEFAULT_OPTIONS: Options = {
     useTabs: false
   },
   unreachableDefinitions: false,
-  unknownAny: true
+  unknownAny: true,
+  enableBigInt: false // True would be more sensible, but also a breaking change from older versions.
```
Comment (bcherny) [inline src/index.ts:112]: Nit: Alphabetize please. Also no need for the comment.

```suggestion
  enableBigInt: false
```

**Review by bcherny — CHANGES_REQUESTED** [review]
Thanks for picking this up! A few stylistic suggestions -- the rest looks good.

File: src/parser.ts (hunk 1 above)
Comment (tristanmenzel) [reply src/parser.ts:48]: options needs to be passed in so that the matchers can read the `enableBigInt`  option

**Comment by bcherny** [comment]
Closing out stale PRs.

**Pull request closed without merging**